### PR TITLE
backend/refactor: changed the payment to KV

### PR DESCRIPTION
**Pull request by enigma-commits**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates

## Description
Changed the payment from persistent to beam. 

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/lib/payment/src/Lib/Payment/Storage/Queries/PaymentOrder.hs [hunk 1]
```diff
@@ -11,57 +11,74 @@
...
-
-updateStatus :: DOrder.PaymentOrder -> SqlDB ()
+import qualified Lib.Payment.Storage.Beam.PaymentOrder as BeamPO
+import qualified Sequelize as Se
+
+findById :: MonadFlow m => Id DOrder.PaymentOrder -> m (Maybe DOrder.PaymentOrder)
+findById (Id paymentOrder) = findOneWithKV [Se.Is BeamPO.id $ Se.Eq paymentOrder]
+
+findByShortId :: MonadFlow m => ShortId DOrder.PaymentOrder -> m (Maybe DOrder.PaymentOrder)
+findByShortId (ShortId shortId) = findAllWithOptionsKV [Se.Is BeamPO.shortId $ Se.Eq shortId] (Se.Desc $ BeamPO.createdAt) (Just 1) Nothing <&> listToMaybe
```
Comment (hkmangla) [inline Backend/lib/payment/src/Lib/Payment/Storage/Queries/PaymentOrder.hs:32]: This query is not correct.

File: Backend/lib/payment/src/Lib/Payment/Storage/Queries/PaymentOrder.hs [hunk 2]
```diff
@@ -11,57 +11,74 @@
...
+updateStatusToExpired (Id orderId) = do
   now <- getCurrentTime
-  Esq.update $ \tbl -> do
-    set
-      tbl
-      [ PaymentOrderStatus =. val Payment.CLIENT_AUTH_TOKEN_EXPIRED,
-        PaymentOrderUpdatedAt =. val now
-      ]
-    where_ $ tbl ^. PaymentOrderId ==. val orderId.getId
+  updateWithKV [Se.Set BeamPO.updatedAt now] [Se.Is BeamPO.id $ Se.Eq orderId]
```
Comment (hkmangla) [inline Backend/lib/payment/src/Lib/Payment/Storage/Queries/PaymentOrder.hs:48]: This is also, incorrect. Lets recheck all queries.

**Review by hkmangla — CHANGES_REQUESTED** [review]

File: Backend/lib/payment/src/Lib/Payment/Storage/Queries/PaymentOrder.hs (hunk 2 above)
Comment (enigma-commits) [reply Backend/lib/payment/src/Lib/Payment/Storage/Queries/PaymentOrder.hs:48]: fixed it

File: Backend/lib/payment/src/Lib/Payment/Storage/Queries/PaymentTransactionDriver.hs
Comment (hkmangla) [inline Backend/lib/payment/src/Lib/Payment/Storage/Queries/PaymentTransactionDriver.hs:1]: Why did we create two file one for each module?

**Pull request closed without merging**